### PR TITLE
Reuse and unify post and page actions, accross the different use cases.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Supersedes: https://github.com/WordPress/gutenberg/pull/59849

This PR introduces a post actions component in the editor package that implements the post and page actions UI in a single place (editor packages):
And reuses the actions in the following scenarios:
- Edit site details view (until now actions only a custom trash action implementation was available there).
- Edit site page inspector (until now actions were not available there).
- Edit post inspector (until now actions were not available there).

This PR makes the PageActions component unnecessary and removes it.
With the merge of https://github.com/WordPress/gutenberg/pull/60395 we will also remove TemplateActions.

## Screenshots
Screenshot 2024-03-13 at 21 24 18
Screenshot 2024-03-13 at 21 25 11
Screenshot 2024-03-13 at 21 31 38

### Testing Instructions for Keyboard
Verified in all the cases where the page actions are used that they work as expected.

### CODE REVIEW

File: packages/editor/src/components/post-actions/index.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+						items={ [ item ] }
+						closeModal={ () => setIsModalOpen( false ) }
+					/>
+				</Modal>
+			) }
+		</>
+	);
+}
+
+// Copied as is from packages/dataviews/src/utils.js
```
Comment (aaronrobertshaw) [inline packages/editor/src/components/post-actions/index.js]:
```suggestion
// Copied as is from packages/dataviews/src/view-table.js
```

I might be missing some related changes but it doesn't appear this function currently lives in `utils.js` and is instead found in `view-table.js`

**Review by aaronrobertshaw — APPROVED** [review]
This is looking great @jorgefilipecosta, nice work! 👍 

It tests well for me. I only spotted a couple of very minor styling issues and tweak for an inline comment.

When a page is in the trash and the page actions button in the site editor's sidebar nav screen is disabled, the hover styles for the button icon seem off.

Screenshot 2024-04-08 at 6 33 24 PM

In the sidebar nav screen still but for a page that isn't in the trash, the icon also appears to disappear be the button is clicked. Is that by design?

https://github.com/WordPress/gutenberg/assets/60436221/f58584ae-bf9a-4291-a613-a8b170f35455

These tweaks could be done in a follow-up, if we want them, so as to not block this PR.

**Comment by jorgefilipecosta** [comment]
Hi @aaronrobertshaw, the active style has been resolved, and the PR is available at https://github.com/WordPress/gutenberg/pull/60592.

Regarding the hover issue, I have observed that we can no longer open a trashed page on the editor, so it seems to not be a problem anymore.  I have added a disabled attribute to the edit button, for testing and unfortunately it has the same issue, so it seems like a general style problem with buttons on dark background.

**Comment by aaronrobertshaw** [comment]
> Regarding the hover issue, I have observed that we can no longer open a trashed page on the editor

Good point. I forgot that improvement was in the works. I can confirm also it is not an issue anymore.

File: packages/editor/src/components/post-actions/index.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	DropdownMenuItemV2: DropdownMenuItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+	DropdownMenuSeparatorV2: DropdownMenuSeparator,
+	kebabCase,
+} = unlock( componentsPrivateApis );
+
+const POST_ACTIONS_WHILE_EDITING = [
+	'view-post',
+	'view-post-revisions',
+	'rename-post',
```
Comment (Mamaduka) [inline packages/editor/src/components/post-actions/index.js:34]: The `rename-post` action is probably redundant for standard post types. The title is available in Canvas, and if the post type doesn't support the `title` field, then renaming should be disabled in the editor.

**Pull request merged**